src/main.rs

```diff
@@ -123,7 +171,7 @@ fn main() {
                 reserved0: 0,
                 tx_ready: 0,
                 mode: UartMode::new(DataBits::Eight, Parity::None, StopBits::One),
-                baud_rate: baud,
+                baud_rate: correct_baud_rate,
```

Comment:
This is incorrect, it sends a configuration update to u-blox.  This should be updating to the requested baud rate and then MUST re-configure the serial port with the new baud rate to confirm it updated as requested.
When not requesting a new baud rate, we should confirm if this parameter is required or not.